### PR TITLE
[Ready for Core Team review] CRM-17463 re-work Group Gontact Cache clause

### PR DESCRIPTION
**Pull request by seamuslee001**
This should mean that we clear the cache table on a group by group basis when no groups are specified in the groupsID param and also we actually set a cache_date

---
- [CRM-17463: Deadlocks on group_cache clearing](https://issues.civicrm.org/jira/browse/CRM-17463)

### CODE REVIEW

**Comment by seamuslee001** [comment]
Jenkins re test this please

**Comment by seamuslee001** [comment]
Jenkins re test this please

File: CRM/Contact/BAO/GroupContactCache.php [hunk 1]
```diff
@@ -331,27 +331,36 @@ public static function remove($groupID = NULL, $onceOnly = TRUE) {
...
+             WHERE group_id = %1";
+          $queryParams = array(1 => array($dao->id, 'Integer'));
+          CRM_Core_DAO::executeQuery($query, $queryParams);
+        }
+        $groups = implode(', ', $groupIds);
         $update = "
-UPDATE civicrm_group g
-SET    cache_date = null,
+UPDATE civicrm_group
+SET    cache_date = $now,
```
Comment (totten) [inline CRM/Contact/BAO/GroupContactCache.php]: I don't think this right. $now hasn't been escaped.

Comment (totten) [inline CRM/Contact/BAO/GroupContactCache.php]: OK, my comment seems to be wrong. `$now` is technically numeric (YmdHis), and MySQL CLI accepts `cache_date` in that format, so maybe it doesn't need escaping.

**Comment by eileenmcnaughton** [comment]
Seamus - I just can't decide if this is a good idea or not. It seems to me that it is a good idea for large groups (reduce the number of deletes in one operation) but  it also changes 1 big query in a large number of small queries.

I'm pretty sure you have gone to production with it - so any feedback about reduction in victim queries could be good - but in general I'm inclined to keep this patch out for this release cycle and get more feedback from you / read up on some of the other proposed improvements in this area first.

**Comment by eileenmcnaughton** [comment]
I propose closing this in favour of https://github.com/civicrm/civicrm-core/pull/8370/files

In terms of this fix, my concern is that it would only be faster in some cases. However, a reasonable extension to https://github.com/civicrm/civicrm-core/pull/8370/files would be for sites that are having problems to use 'deterministic' mode & potentially enhance that by adding more flexibility from the batch job end -e.g some of

1) a truncate batch job option - for speed
2) potentially a groups filter on the batch job
3) a more complex batch job that finds the largest expired groups & does them first.

**Comment by eileenmcnaughton** [comment]
Closed in favour of https://github.com/civicrm/civicrm-core/pull/8370

**Pull request closed without merging**